status/health/window/error_source.go

```diff
@@ -143,9 +154,8 @@ func newHealthyIfNotAllErrorsSource(checkType health.CheckType, windowSize time.
 		checkType:          checkType,
 	}
 
-	if useAnchoredWindows {
-		retVal.lastSuccessTime = retVal.timeProvider.Now()
-	}
+	// Always require the first initial window to elapse before reporting unhealthy
+	retVal.lastSuccessTime = retVal.timeProvider.Now()
```

Comment:
i thought we discussed that we won't do this via using the last success "hack" but instead via a first class that like "startUpTime" or something like that.
not super strongly opposed to this but this is again way too magical imo.